Index: package/base-files/files/etc/init.d/boot
===================================================================
--- package/base-files/files/etc/init.d/boot.orig	2016-02-28 13:34:20.701293566 +0100
+++ package/base-files/files/etc/init.d/boot	2016-02-28 13:34:20.697293566 +0100
@@ -31,8 +31,6 @@
 	mkdir -p /tmp/.jail
 	touch /var/log/wtmp
 	touch /var/log/lastlog
-	touch /tmp/resolv.conf.auto
-	ln -sf /tmp/resolv.conf.auto /tmp/resolv.conf
 	grep -q debugfs /proc/filesystems && /bin/mount -o noatime -t debugfs debugfs /sys/kernel/debug
 	[ "$FAILSAFE" = "true" ] && touch /tmp/.failsafe
 
Index: package/network/services/dnsmasq/files/dnsmasq.init
===================================================================
--- package/network/services/dnsmasq/files/dnsmasq.init.orig	2016-02-28 13:34:20.701293566 +0100
+++ package/network/services/dnsmasq/files/dnsmasq.init	2016-02-28 13:34:20.697293566 +0100
@@ -634,7 +634,6 @@
 stop_service() {
 	[ -f /tmp/resolv.conf ] && {
 		rm -f /tmp/resolv.conf
-		ln -s /tmp/resolv.conf.auto /tmp/resolv.conf
 	}
 	rm -f /var/run/dnsmasq.*.dhcp
 }
