Index: openwrt/target/linux/generic/config-2.6.37
===================================================================
--- target.orig/linux/generic/config-2.6.37	2011-06-19 16:28:32.000000000 +0200
+++ target/linux/generic/config-2.6.37	2011-06-19 16:28:38.000000000 +0200
@@ -1621,7 +1621,7 @@
 # CONFIG_NF_CONNTRACK_IPV4 is not set
 # CONFIG_NF_CONNTRACK_IPV6 is not set
 # CONFIG_NF_CONNTRACK_IRC is not set
-# CONFIG_NF_CONNTRACK_MARK is not set
+CONFIG_NF_CONNTRACK_MARK=y
 # CONFIG_NF_CONNTRACK_NETBIOS_NS is not set
 # CONFIG_NF_CONNTRACK_PPTP is not set
 # CONFIG_NF_CONNTRACK_PROC_COMPAT is not set
