Index: openwrt/target/linux/generic/config-2.6.36
===================================================================
--- openwrt.orig/target/linux/generic/config-2.6.36	2011-06-19 20:53:44.000000000 +0200
+++ openwrt/target/linux/generic/config-2.6.36	2011-06-19 20:54:31.000000000 +0200
@@ -1827,7 +1827,7 @@
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_PRINTK=y
-# CONFIG_PRINTK_TIME is not set
+CONFIG_PRINTK_TIME=y
 # CONFIG_PRISM2_USB is not set
 # CONFIG_PRISM54 is not set
 # CONFIG_PROBE_INITRD_HEADER is not set
Index: openwrt/target/linux/generic/config-2.6.37
===================================================================
--- openwrt.orig/target/linux/generic/config-2.6.37	2011-06-19 20:54:45.000000000 +0200
+++ openwrt/target/linux/generic/config-2.6.37	2011-06-19 20:54:53.000000000 +0200
@@ -1870,7 +1870,7 @@
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_PRINTK=y
-# CONFIG_PRINTK_TIME is not set
+CONFIG_PRINTK_TIME=y
 # CONFIG_PRISM2_USB is not set
 # CONFIG_PRISM54 is not set
 # CONFIG_PROBE_INITRD_HEADER is not set
Index: openwrt/target/linux/generic/config-2.6.38
===================================================================
--- openwrt.orig/target/linux/generic/config-2.6.38	2011-06-19 20:54:57.000000000 +0200
+++ openwrt/target/linux/generic/config-2.6.38	2011-06-19 20:55:03.000000000 +0200
@@ -1902,7 +1902,7 @@
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_PRINTK=y
-# CONFIG_PRINTK_TIME is not set
+CONFIG_PRINTK_TIME=y
 # CONFIG_PRISM2_USB is not set
 # CONFIG_PRISM54 is not set
 # CONFIG_PROBE_INITRD_HEADER is not set
Index: openwrt/target/linux/generic/config-2.6.39
===================================================================
--- openwrt.orig/target/linux/generic/config-2.6.39	2011-06-19 20:55:10.000000000 +0200
+++ openwrt/target/linux/generic/config-2.6.39	2011-06-19 20:55:17.000000000 +0200
@@ -1938,7 +1938,7 @@
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_PRINTK=y
-# CONFIG_PRINTK_TIME is not set
+CONFIG_PRINTK_TIME=y
 # CONFIG_PRISM2_USB is not set
 # CONFIG_PRISM54 is not set
 # CONFIG_PROBE_INITRD_HEADER is not set
Index: openwrt/target/linux/generic/config-3.0
===================================================================
--- openwrt.orig/target/linux/generic/config-3.0	2011-06-19 20:55:21.000000000 +0200
+++ openwrt/target/linux/generic/config-3.0	2011-06-19 20:55:27.000000000 +0200
@@ -1943,7 +1943,7 @@
 # CONFIG_PREEMPT_VOLUNTARY is not set
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_PRINTK=y
-# CONFIG_PRINTK_TIME is not set
+CONFIG_PRINTK_TIME=y
 # CONFIG_PRISM2_USB is not set
 # CONFIG_PRISM54 is not set
 # CONFIG_PROBE_INITRD_HEADER is not set
