Index: target/linux/ar71xx/files/drivers/mtd/tplinkpart.c
===================================================================
--- target/linux/ar71xx/files/drivers/mtd/tplinkpart.c.orig	2014-11-14 14:14:31.657150281 +0100
+++ target/linux/ar71xx/files/drivers/mtd/tplinkpart.c	2014-11-14 14:14:42.785149863 +0100
@@ -149,7 +149,6 @@
 	parts[0].name = "u-boot";
 	parts[0].offset = 0;
 	parts[0].size = offset;
-	parts[0].mask_flags = MTD_WRITEABLE;
 
 	parts[1].name = "kernel";
 	parts[1].offset = offset;
