Index: target/linux/generic/files/drivers/net/phy/adm6996.c
===================================================================
--- target/linux/generic/files/drivers/net/phy/adm6996.c
+++ target/linux/generic/files/drivers/net/phy/adm6996.c
@@ -1100,4 +1100,9 @@
 }
 
+static int adm6996_soft_reset(struct phy_device *phydev)
+{
+	/* we don't need an extra reset */
+	return 0;
+}
 
 static struct phy_driver adm6996_phy_driver = {
@@ -1111,4 +1116,5 @@
 	.config_aneg	= &adm6996_config_aneg,
 	.read_status	= &adm6996_read_status,
+	.soft_reset	= adm6996_soft_reset,
 	.driver		= { .owner = THIS_MODULE,},
 };
