--- ntpclient-2007.orig/ntpclient.c	2009-09-20 15:11:21.000000000 +0200
+++ ntpclient-2007/ntpclient.c	2009-09-20 15:11:52.000000000 +0200
@@ -588,7 +588,7 @@
 #ifdef ENABLE_DEBUG
 	" [-d]"
 #endif
-	" [-f frequency] [-g goodness] -h hostname\n"
+	" [-D] [-f frequency] [-g goodness] -h hostname\n"
 	"\t[-i interval] [-l] [-p port] [-q min_delay]"
 #ifdef ENABLE_REPLAY
 	" [-r]"
@@ -614,7 +614,7 @@
 	ntpc.cross_check=1;
 
 	for (;;) {
-		c = getopt( argc, argv, "c:" DEBUG_OPTION "f:g:h:i:lp:q:" REPLAY_OPTION "st");
+		c = getopt( argc, argv, "c:" DEBUG_OPTION "f:g:h:i:lp:q:" REPLAY_OPTION "stD");
 		if (c == EOF) break;
 		switch (c) {
 			case 'c':
@@ -663,6 +663,10 @@
 				(ntpc.cross_check)=0;
 				break;
 
+			case 'D':
+				daemon();
+				break;
+
 			default:
 				usage(argv[0]);
 				exit(1);
